### ISSUE TITLE
Cannot hide or remove snackbar displayed by OneContext().showSnackBar

### ISSUE THREAD

**Issue by mattbreeland** [emanuel-braz/one_context#19]
I want snackbars to disappear if the user clicks anywhere on them, but am unable to make it work with OneContext.

```
OneContext().showSnackBar(
    builder: (context) => SnackBar(
      content: GestureDetector(
        onTap: () {
          ScaffoldMessenger.of(context!).removeCurrentSnackBar(); //hideCurrentSnackBar also fails
        },
        child: Container(/* snip */),
    ),
  );
```

Is there another context object I should use? Or is there some other workaround to enable this?

**Comment by mattbreeland** [comment]
Note:

```
onTap: () {
          OneContext().hideCurrentSnackBar();
        },
```
works fine, but it is deprecated! What should I use instead?

**Comment by emanuel-braz** [comment]
> I want snackbars to disappear if the user clicks anywhere on them, but am unable to make it work with OneContext.
> 
> ```
> OneContext().showSnackBar(
>     builder: (context) => SnackBar(
>       content: GestureDetector(
>         onTap: () {
>           ScaffoldMessenger.of(context!).removeCurrentSnackBar(); //hideCurrentSnackBar also fails
>         },
>         child: Container(/* snip */),
>     ),
>   );
> ```
> 
> Is there another context object I should use? Or is there some other workaround to enable this?

I have to check it out, but I believe you can use the `OneContext().context`. And, if needed, you can check if the context is available, using `OneContext.hasContext`

**Comment by emanuel-braz** [comment]
> Note:
> 
> ```
> onTap: () {
>           OneContext().hideCurrentSnackBar();
>         },
> ```
> 
> works fine, but it is deprecated! What should I use instead?

You are right. And actually, I need to migrate the lib in order to use ScaffoldMessenger.

ref.
https://flutter.dev/docs/release/breaking-changes/scaffold-messenger

**Comment by emanuel-braz** [comment]
Issue fixed with PR #20
